packages/cli-lib/lib/handleFieldsJs.js

```diff
@@ -0,0 +1,218 @@
+const fs = require('fs-extra');
+const os = require('os');
+const path = require('path');
+const escapeRegExp = require('./escapeRegExp');
+const yargs = require('yargs');
+const { splitLocalPath } = require('../path');
+const { logger } = require('../logger');
+const { getCwd, getExt } = require('@hubspot/cli-lib/path');
+const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const i18nKey = 'cli.commands.upload';
+
+/**
+ * FieldsJS Class.
+ * @param {string} projectDir - The root directory of the filePath
+ * @param {string} filePath - The path to the fields.js file to be converted
+ * @param {string} [rootWriteDir] - (Optional) The root of the directory in which to output the fields.js. If blank, a temporary directory is created.
+ */
+class FieldsJs {
+  constructor(projectDir, filePath, rootWriteDir) {
+    this.projectDir = projectDir;
+    this.filePath = filePath;
+
+    // Create tmpDir if no writeDir is given.
+    this.rootWriteDir =
+      rootWriteDir === undefined
+        ? createTmpDirSync('hubspot-temp-fieldsjs-output-')
+        : rootWriteDir;
+  }
+
+  /**
+   * Converts a fields.js file into a fields.json file, writes, and returns of fields.json
+   * @param {string} file - The path of the fields.js javascript file.
+   * @param {string} writeDir - The directory to write the file to.
+   * @returns {Promise} finalPath - Promise that returns path of the written fields.json file.
+   */
+  convertFieldsJs(writeDir) {
+    const filePath = this.filePath;
+    const options = yargs.argv.options;
```

Comment:
instead of getting `options` directly from `yargs` can we pass it down from the handler?

Reply:
Yes. I think I'll rename `options` to `fieldOptions` as well. Current naming is confusing.